### PR TITLE
Add internal Java solution for Taming the Herd + minor fix of old file 1060

### PR DESCRIPTION
**Pull request by sathu-beep**

### CODE REVIEW

File: solutions/usaco-809.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
+
+<LanguageSection>
+
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+class Main {
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:17]:
```suggestion

public class Taming {
```

File: solutions/usaco-1060.mdx [hunk 2]
```diff
@@ -61,8 +61,8 @@ import java.util.*;
 import java.io.*;
 	
 class Main {
-	public static void main(String[] args) {
-		Kattio io = new Kattio()
+	public static void main(String[] args) throws IOException{
```
Comment (HarryWangATX) [inline solutions/usaco-1060.mdx:64]: why throw the exception? im p sure console io doesnt need it.

File: solutions/usaco-809.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			io.println(-1);
+			io.close();
+			System.exit(0);
+		}
+		log[0] = 0;
+		int t = -1;
+		int req = 0;
+		int pos = 0;
+		for(int i = n - 1; i >= 0; i--){
+			if( t != -1 && log[i] != -1 && log[i] != t){
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:35]: what is this spacing sathu lmao

File: solutions/usaco-809.mdx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+---
+id: usaco-809
+source: USACO Bronze 2018 Febuary
+title: Taming the Herd
+author: Sathvik Chundru
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
+
```
Comment (HarryWangATX) [inline solutions/usaco-809.mdx:9]: time comple

File: solutions/usaco-809.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-809.mdx:35]: space your control statements like this
```java
for ([insert stuff here]) {
}
if ([condition]) {
}
```

File: solutions/usaco-809.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+				req++;
+			}
+			if(log[i] == -1){
+				pos++;
+			}
+			if(t > -1){
+				t--;
+			}
+		}
+	int sum = req + pos;
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:56]: indentation's messed up

File: solutions/usaco-809.mdx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+			log[i] = io.nextInt();
+		}
+		if(log[0] > 0){
+			io.println(-1);
+			io.close();
+			System.exit(0);
+		}
+		log[0] = 0;
+		int t = -1;
+		int req = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:32]: comments would be nice

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-809.mdx (hunk 1 above)
Comment (HarryWangATX) [inline solutions/usaco-809.mdx:17]:
```suggestion
public class Taming{
```

File: solutions/usaco-809.mdx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+				pos++;
+			}
+			if(t > -1){
+				t--;
+			}
+		}
+	int sum = req + pos;
+	io.println(req + " " + sum);
+	io.close();
+	}
```
Comment (HarryWangATX) [inline solutions/usaco-809.mdx:59]: indentation kinda sus

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/usaco-1060.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-1060.mdx:64]: think it does?

Comment (sathu-beep) [reply solutions/usaco-1060.mdx:64]: it does need it it wont work without it

Comment (HarryWangATX) [reply solutions/usaco-1060.mdx:64]: i just checked, u do not need it

Comment (nathangong) [reply solutions/usaco-1060.mdx:64]: ^^ I can confirm, Kattio without file i/o doesn't need a `throws` statement

File: solutions/usaco-809.mdx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+<LanguageSection>
+
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class Taming{
+	public static void main(String[] args) throws IOException {
```
Comment (nathangong) [inline solutions/usaco-809.mdx:19]: Either add spacing to both of these before the beginning of the curly brackets or remove them

File: solutions/usaco-809.mdx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+		int t = -1;
+		int req = 0;
+		int pos = 0;
+		for(int i = n - 1; i >= 0; i--){
+			if( t != -1 && log[i] != -1 && log[i] != t){
+				io.println(-1);
+				io.close();
+				System.exit(0);
+			}
+			if(t == -1){
```
Comment (nathangong) [inline solutions/usaco-809.mdx:41]: Can you add an explanation for this long pile of if statements? It's not super obvious at first glance what the purpose of all of these are.

File: solutions/usaco-809.mdx (hunk 1 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: thanks

File: solutions/usaco-809.mdx (hunk 3 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx (hunk 5 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx:35]: the official cpp solution explains it better in paragraph form so no need to make code ugly by adding comments that will never do as good a job of explaining the problem as the official solution. Mine is just for people who do java so they can understand what is going on in the cpp code.

File: solutions/usaco-809.mdx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+		if( log[0] > 0 ){
+			io.println(-1);
+			io.close();
+			System.exit(0);
+		}
+		log[0] = 0;
+		int t = -1;
+		int req = 0;
+		int pos = 0;
+		for( int i = n - 1; i >= 0; i-- ){
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:37]: :notlikethis:
i mean something like
```java
for (int i = n - 1; i >= 0; i--) {
}
```

File: solutions/usaco-1060.mdx (hunk 2 above)
Comment (sathu-beep) [reply solutions/usaco-1060.mdx]: removed throws IOE exception but i still need to add the ";"

File: solutions/usaco-809.mdx (hunk 4 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx:9]: added

File: solutions/usaco-809.mdx (hunk 7 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx (hunk 8 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx (hunk 9 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: addressed this above

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-809.mdx:35]: ...then copy paste the comments?

Comment (sathu-beep) [reply solutions/usaco-809.mdx:35]: its like 3 paragraphs above the code. If you want me to I can include it in mine

File: solutions/usaco-809.mdx (hunk 10 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-809.mdx:35]: if you mean it's included in the editorial, i'd suggest you include the relevant parts of it in this code so people know which code segment matches to which editorial sentence
(it'd be cool if you could do this for the cpp version as well)
but if there's a editorial i wouldn't really care if you included it or not

File: solutions/usaco-809.mdx (hunk 10 above)
Comment (SansPapyrus683) [reply solutions/usaco-809.mdx]: did you not see the spaces here
image
image

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx:35]: I will try later

File: solutions/usaco-809.mdx (hunk 10 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx]: fixed

File: solutions/usaco-809.mdx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class Taming {
+	public static void main(String[] args) {
+		Kattio io = new Kattio("taming");
+		int n = io.nextInt();
+		int[] log = new int[100000];
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:24]: java can have dynamically sized arrays
`int[] log = new int[n];` instead?

Comment (sathu-beep) [reply solutions/usaco-809.mdx:24]: fixed

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx:42]: fixed @HarryWangATX @dongliu0426 if its good merge now

Comment (sathu-beep) [reply solutions/usaco-809.mdx:42]: @SansPapyrus683

File: solutions/usaco-809.mdx [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+		 * Otherwise, the log must consist of several streaks 0,1,2,…,k 
+		 * of various lengths, with possibly some −1s between 
+		 * streaks - entries which we could not uniquely deduce. 
+		 * We know that the first streak starts on the first day.
+		 */
+		for (int i = n - 1; i >= 0; i--) {
+			if (t != -1 && log[i] != -1 && log[i] != t) {
+				io.println(-1);
+				io.close();
+				System.exit(0);
```
Comment (nwatx) [inline solutions/usaco-809.mdx:56]: use `break`?

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/usaco-809.mdx [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+			}
+			if (log[i] == -1) {
+				log[i] = t;
+			}
+			if (log[i] == 0) {
+				req++;
+			}
+			if (log[i] == -1) {
+				pos++;
+			}
```
Comment (nwatx) [inline solutions/usaco-809.mdx:69]: could some of these ifs be combined?

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/usaco-809.mdx (hunk 13 above)
Comment (sathu-beep) [reply solutions/usaco-809.mdx:69]: not really since they all modify separate things?

Comment (ar492) [reply solutions/usaco-809.mdx:69]: probably not

File: solutions/usaco-809.mdx [hunk 14]
```diff
@@ -25,15 +25,30 @@ public class Taming {
...
+		 * since Farmer John already knows the cows broke out on that day.
+		 */
 		if (log[0] > 0) {
 			io.println(-1);
 			io.close();
 			System.exit(0);
 		}
+		/*
+		 *  If it is a missing entry, we can set it to 0.
+		 */
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:39]: no need for a multiline comment here

Comment (sathu-beep) [reply solutions/usaco-809.mdx:39]: fixed

**Review by SansPapyrus683 — APPROVED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
not too awful

**Comment by SansPapyrus683** [comment]
wait crap forgot i approved sorry lol

**Review by darren-yao — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**